### DISCUSSION TITLE
Help with email delivery service not working and with web hooks variables

### DISCUSSION THREAD

**Discussion by rodrigo2k** [AzuraCast/AzuraCast#7485] [Q&A] [answered]
Greetings fellow AzuraCast users,

Could you please advise me on the following? I have set up the SMTP settings for email delivery on one of my servers, and test emails or any emails, for that matter, never get delivered. I can confirm all the account settings are correct because I use the same on a separate AzuraCast server, which works without any delays in receiving the emails.

I don't know what logs to check to see what the problem could be. I am confident enough to run docker commands in my Linux VPS shell, but I am truly lost at what to do.

On the server where emails are working, can someone please show me an example of using the webhook variables from the NowPlaying API? I lack understanding on how to use the variables displayed in this sample page https://demo.azuracast.com/api/nowplaying_static/azuratest_radio.json

![image](https://github.com/user-attachments/assets/166c9ad3-70ef-4b93-82d0-85a58bfe6112)

Much appreciated!

**Comment by LordHelmchen666** [comment, marked as answer]
Webhooks usage is very easy. 
Station Name = {{ station.name }}
Listener Total = {{ listeners.total }}
now playing artist and title =  {{ now_playing.song.text }}
Hope the scheme is clear now.

Did you check logs on the server where your mail account is hosted? Are you sure you are not blocked? Could you connect directly via console from your Azuracast server to mail server? e.g. via telnet https://www.thomas-krenn.com/en/wiki/Test_TCP_Port_25_(smtp)_access_with_telnet

**Reply by LordHelmchen666** [reply]
If you have an SPF record set, do not forget to add your Azuracast server IP to allowed IPs.
Example 
`v=spf1 +a +mx +ip4:8.8.8.8 +ip4:9.9.9.9 +a:mydommain.com -all`
Explanation can you find here https://dmarcian.com/spf-syntax-table/

**Reply by rodrigo2k** [reply]
Dear @LordHelmchen666 thank you, thank you, thank you!

I now have both "issues" resolved thanks to your help, the emails were not being delivered because my host had the SMTP port blocked as part of their spam preventing measures, this was removed once I reached out to them.

Thank you too for providing help with the codes to use in the emails, it is a pitty the body of the email is a one liner and it cannot be modified with formatting, unless I don't know how to use this??? the plot thickens.

Cheers!
